### ISSUE TITLE
Configmap and application in other namespace then Spring Cloud Kubernetes Configuration Watcher

### ISSUE THREAD

**Issue by TimMickelson** [spring-cloud/spring-cloud-kubernetes#1573]
**Spring Cloud Kubernetes Configuration Watcher and Spring Boot POD with configmap in namespace A**
I have a working setup with Spring Cloud Kubernetes Configuration Watcher in namespace A configured with RabbitMQ. I have a microservice in the same namespace and if I edit the configmap the application context is refreshed as expected.

**Spring Boot POD with configmap in namespace B**
I have a POD in namespace B, I have configured RBAC so the watcher has permissions to receive events from the corresponding configmap. When I change the configmap no event seems to arrive at the watcher and so, the application does not get notification to refresh application context.

So my question is if this is a possible setup, and if so, is there some configuration that I must set, and if so, what configuration? It seems to be wrong to install a watcher on each namespace.

**Comment by TimMickelson** [comment]
I can confirm that this change make the watcher see changes on namespace resesok-paratransit, but now I don't see changes on namespace otp (namespace A in my question). I tried adding bouth namespaces in the value but that made the watcher crash.

**Comment by TimMickelson** [comment]
I can verify that this works well for refreshing in both namespaces. The output on the log in the watcher is not very beautiful, it give the same error as I saw before, with the difference that it updates well. Should I configure the error out of my log?

```
2024-02-07T11:47:02.785Z ERROR 1 --- [dels.V1Secret-1] i.k.c.informer.cache.ReflectorRunnable   : class io.kubernetes.client.openapi.models.V1Secret#Reflector loop failed unexpectedly

io.kubernetes.client.openapi.ApiException: 
	at io.kubernetes.client.openapi.ApiClient.handleResponse(ApiClient.java:973) ~[client-java-api-17.0.2.jar:na]
	at io.kubernetes.client.openapi.ApiClient.execute(ApiClient.java:885) ~[client-java-api-17.0.2.jar:na]
	at io.kubernetes.client.informer.SharedInformerFactory$1.list(SharedInformerFactory.java:222) ~[client-java-17.0.2.jar:na]
	at io.kubernetes.client.informer.cache.ReflectorRunnable.run(ReflectorRunnable.java:91) ~[client-java-17.0.2.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[na:na]
	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
```

**Comment by TimMickelson** [comment]
I will do that, for me this is good enough because it is working. To bad about the logs, but they don't seem to be a show stopper for me. I will create an issue here and if it is prioritized someone can fix it in future release.

All your help is very appreciated you fixed my problem.

**Comment by TimMickelson** [comment]
Little comment, if I go to spring.io then I end up on different documentation from the one you linked me to.

[spring.io](https://docs.spring.io/spring-cloud-kubernetes/reference/index.html)

You gave me [this](https://docs.spring.io/spring-cloud-kubernetes/docs/current/reference/html/#spring-cloud-kubernetes-configuration-watcher)

**Comment by TimMickelson** [comment]
Created issue #1574

**Comment by ryanjbaxter** [comment]
These are just different versions of the documentation for different releases.  The version on spring.io you linked to is for the 3.1.0 release the version @wind57 pointed to is for the 3.0.x release.  The content for the section on the configuration watcher should be virtually the same between the two releases I believe.

**Comment by TimMickelson** [comment]
I understand, but if the configuration is still valid, then I think it should remain in the documentation of 3.1.0?

**Comment by ryanjbaxter** [comment]
Ah those documentation changes were merged into the 3.1.x branch (main) after the 3.1.0 release.  They are there in our 3.1.1 snapshot documentation and will be published with the next release of that branch
https://docs.spring.io/spring-cloud-kubernetes/reference/3.1-SNAPSHOT/spring-cloud-kubernetes-configuration-watcher.html

**Comment by TimMickelson** [comment]
This makes more sense, thanks for the feedback.
